# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/api/FileSystemManager.cpp
# Commit: 214c4d6bf7f9
# Full Hash: 214c4d6bf7f98a332c4ab81d325d9245ae810955
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1792245
# File Overlap Count: 2
# Description:
#   Bug 1792245 - Add ErrorResult to FileSystemRequestHandler methods; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159572
# ==============================================================================

diff -r 330c69218a93 -r 214c4d6bf7f9 dom/fs/api/FileSystemManager.cpp
--- a/dom/fs/api/FileSystemManager.cpp	Thu Oct 20 07:33:41 2022 +0300
+++ b/dom/fs/api/FileSystemManager.cpp	Thu Oct 20 04:21:30 2022 +0000
@@ -102,17 +102,20 @@
              });
 }
 
-already_AddRefed<Promise> FileSystemManager::GetDirectory(ErrorResult& aRv) {
+already_AddRefed<Promise> FileSystemManager::GetDirectory(ErrorResult& aError) {
   MOZ_ASSERT(mGlobal);
 
-  RefPtr<Promise> promise = Promise::Create(mGlobal, aRv);
-  if (NS_WARN_IF(aRv.Failed())) {
+  RefPtr<Promise> promise = Promise::Create(mGlobal, aError);
+  if (NS_WARN_IF(aError.Failed())) {
     return nullptr;
   }
 
   MOZ_ASSERT(promise);
 
-  mRequestHandler->GetRootHandle(this, promise);
+  mRequestHandler->GetRootHandle(this, promise, aError);
+  if (NS_WARN_IF(aError.Failed())) {
+    return nullptr;
+  }
 
   return promise.forget();
 }